### ISSUE TITLE
Cypress server StatusCodeError: 412 on hookId and hookName is required

### ISSUE THREAD

**Issue by char1e5** [cypress-io/cypress#8288]
### Current behavior:
When we run cypress in parallel, `cypress@4.10.0` and above receives service error during `Uploading Results`. 
[Cypress dashboard](https://dashboard.cypress.io/) will be pending and never finish. 

See the below error messages:

```
  (Uploading Results)

We encountered an unexpected error talking to our servers.

Because you passed the --parallel flag, this run cannot proceed because it requires a valid response from our servers.

The --ciBuildId flag you passed was: xxx

The server's response was:

StatusCodeError: 412

{
  "message": "Request should follow putInstanceRequest@2.1.0 schema",
  "errors": [
    "data.hooks.1.hookId is required",
    "data.hooks.1.hookName is required"
  ],
  "object": {
    "stats": {
      "suites": 8,
      "tests": 16,
      "passes": 16,
      "pending": 0,
      "skipped": 0,
      "failures": 0,
      "wallClockStartedAt": "2020-08-14T20:33:38.948Z",
      "wallClockEndedAt": "2020-08-14T20:34:28.554Z",
      "wallClockDuration": 49606
    },
    "tests": [
      {
        "testId": "r4",
        "title": [
          "Generic Landing Page",
          "Landing Page Components",
          "Should be present on load"
        ],
        "state": "passed",
```

### Desired behavior:
Tested different cypress versions includes `4.9.0`, `4.10.0` and `4.12.1`. `4.9.0` works fine without issues.
Looking at the change log https://github.com/cypress-io/cypress/releases/tag/v4.10.0 
It includes hooks updates. That might cause the server error issue.
We have downgrade cypress from the latest version to `4.9.0` to run tests with cypress server.

### Test code to reproduce
```
cypress run --headless --browser chrome --record --key=<key> --ci-build-id=<build_id> --parallel --config baseUrl=<url>
```

### Versions

`4.10.0` and above

**Comment by jennifer-shehane** [comment]
@char1e5 Can you provide the full error log? It looks like it may be cutting off beyond the beginning of the tests array. 

There's likely a situation where the tests can run and incorrectly create a hook, which has no `hookId` or `hookName`. Is it possible to send over an example of the test you're running - specifically around the structure of the hooks/suite/tests? This would help us recreate the situation where this would happen. Feel free to send anything over to support@cypress.io.

**Comment by jennifer-shehane** [comment]
Hey @char1e5 and @samuelshirley, we have some reason to believe that you may have written some of the tests with hooks nested within hooks, so something like below (or any other defined hook):

```js
beforeEach(() => {
  afterEach(() => {})
})
```

Can either of you confirm that there are nested hooks within your test suite somewhere?

We are still discussing how we want to handle nested hooks in the future (we may error when they are defined altogether) 

For now, we recommend that all hooks be defined separately. This change would not change how the hooks execute.

**Comment by char1e5** [comment]
@jennifer-shehane searching for our code, we do have nested hooks in our tests. But I think that's a valid case we had earlier. We need to define some common hooks to run it before all tests. If we separate the hooks, it's not ideal for us to manage our tests.

**Comment by Bkucera** [comment]
@char1e5 If you want common hooks, you can place them in a `support` file. Could you give an example where a nested hook is required?

**Comment by char1e5** [comment]
@Bkucera yes, we have 2 places having nested hooks in our tests. 
1. In support file, we do have some common hooks, I thought it was considered as `nested hooks` earlier.  But it's really just one common hooks with nested before, which is incorrect code in my opinion. It has `before(() => (before(() => {})))`. 
2. In one test file, we have `beforeEach(() => (beforeEach(() => {})))`. We can change fix it, too.

Thanks to point this out about nested hooks. I'd like to have some cypress alerts or even support nested hooks in the future. It's really hard us to tell what's wrong from the error message. 
At the same time, I can update the code and see how it works in the new version of cypress.

**Comment by lastquestion** [comment]
@jennifer-shehane I suggest erroring, there should be no case where you want to nest hooks that cannot be simplified into flat hooks. Plus it's super confusing to maintain those kinds of tests.

**Comment by Bkucera** [comment]
erroring on nested hooks is implemented in #8400
